### PR TITLE
fix: open redirects

### PR DESCRIPTION
**Pull request by jacob-ebey**

### CODE REVIEW

File: app/routes/join.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const action: ActionFunction = async ({ request }) => {
     request,
     userId: user.id,
     remember: false,
-    redirectTo: typeof redirectTo === "string" ? redirectTo : "/",
+    redirectTo,
```
Comment (MichaelDeBoey) [inline app/routes/join.tsx:69]: Maybe inline the call, so we're first doing redirects before "calculating" it?

```suggestion
    redirectTo: safeRedirect(formData.get("redirectTo"), "/"),
```

File: app/routes/join.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const action: ActionFunction = async ({ request }) => {
   const formData = await request.formData();
   const email = formData.get("email");
   const password = formData.get("password");
-  const redirectTo = formData.get("redirectTo");
+  const redirectTo = safeRedirect(formData.get("redirectTo"), "/");
```
Comment (MichaelDeBoey) [inline app/routes/join.tsx:32]:
```suggestion
```

File: app/routes/login.tsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ export const action: ActionFunction = async ({ request }) => {
     request,
     userId: user.id,
     remember: remember === "on" ? true : false,
-    redirectTo: typeof redirectTo === "string" ? redirectTo : "/notes",
+    redirectTo,
```
Comment (MichaelDeBoey) [inline app/routes/login.tsx:68]: Maybe inline the call, so we're first doing redirects before "calculating" it?

```suggestion
    redirectTo: safeRedirect(formData.get("redirectTo"), "/notes"),
```

File: app/routes/login.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ export const action: ActionFunction = async ({ request }) => {
   const formData = await request.formData();
   const email = formData.get("email");
   const password = formData.get("password");
-  const redirectTo = formData.get("redirectTo");
+  const redirectTo = safeRedirect(formData.get("redirectTo"), "/notes");
```
Comment (MichaelDeBoey) [inline app/routes/login.tsx:31]:
```suggestion
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by kentcdodds — APPROVED** [review]
I'm happy with how this is implemented 👍 Thanks!

**Pull request merged**